sympy/core/tests/test_numbers.py

```diff
@@ -1749,3 +1749,32 @@ def test_Integer_precision():
     assert Float('1.0', precision=Integer(15))._prec == 15
     assert type(Float('1.0', precision=Integer(15))._prec) == int
     assert sympify(srepr(Float('1.0', precision=15))) == Float('1.0', precision=15)
+
+def test_numpy_to_float():
+    from sympy.utilities.pytest import skip
+    from sympy.external import import_module
+    np = import_module('numpy')
+    if not np:
+        skip('numpy not installed. Abort numpy tests.')
+
+    def check_prec_and_relerr(npval, ratval):
+        prec = np.finfo(npval).nmant
+        x = Float(npval)
+        assert x._prec == prec
+        y = Float(ratval, precision=prec)
+        assert abs((x - y)/y) < 2**(-(prec+1))
+
+    check_prec_and_relerr(np.float32(2)/3, S(2)/3)
+    check_prec_and_relerr(np.float64(2)/3, S(2)/3)
+    check_prec_and_relerr(np.float128(2)/3, S(2)/3)
+
+    # float128 more precise than double (how much more is arch dep)
+    x = np.float128(2)/3
```

Comment:
`float128` may not be available exist on all platforms. You should do attribute check or remove it.
Fails on Windows 32-bit: [Appveyor](https://ci.appveyor.com/project/ylemkimon/sympy/build/1.0.16).

Reply:
Thanks for this!  [The docs](https://docs.scipy.org/doc/numpy/user/basics.types.html) say:

>  In spite of the names, np.float96 and np.float128 provide only as much precision
>  as np.longdouble, that is, 80 bits on most x86 machines and 64 bits in standard Windows builds.

So yes this particular test should be removed.

However, the docs also say:

>  np.float96 and np.float128 are provided for users who want specific padding.

But you say `float128` may not be available.  And indeed `np.float96` is not available on my GNU/Linux 64-bit!  I guess I'll change the test to `longdouble`.